packages/auth/src/api/index.ts

```diff
@@ -265,11 +266,21 @@ export function _getFinalTarget(
 ): string {
   const base = `${host}${path}?${query}`;
 
-  if (!(auth as AuthInternal).config.emulator) {
-    return `${auth.config.apiScheme}://${base}`;
+  const finalTarget = (auth as AuthInternal).config.emulator
```

Comment:
Small nit: can we create a `const authInternal = auth as AuthInternal` so that we only have to do that `as` cast once?